### ISSUE TITLE
DSLContext ddl does not correctly export view contents from existing database

### ISSUE THREAD

**Issue by Rosenrot48** [jOOQ/jOOQ#14712]
### Expected behavior

Something like this

```sql
create view "public"."demo_description"("id","name","description")
as (SELECT de.id,
de.name,
ds.text AS description
FROM demo de
JOIN description ds ON ds.id = de.id;)
```

### Actual behavior

```sql
create view "public"."demo_description"(
"id",
"name",
"description"
)
as
```

### Steps to reproduce the problem

1.  Connect Spring boot application to existing DataSource
2. Try to generate DDL information about tables
```java
public void execute() {
        var context = DSL.using(dataSource, SQLDialect.POSTGRES);
        var queries = context.meta().ddl();
        for (Query query: queries) {
            logger.info("Result query: " + query);
        }
    }
````

### jOOQ Version

JOOQ OpenSource Edition 3.17.8

### Database product and version

PostgreSQL 13.3 on x86_64-apple-darwin19.6.0, compiled by Apple clang version 11.0.3 (clang-1103.0.32.62), 64-bit

### Java Version

openjdk 11.0.10 2021-01-19

### OS Version

MacOS 13.0

### JDBC driver name and version (include name if unofficial driver)

org.postgresql:postgresql:42.5.4

Project with issue is available at https://github.com/Rosenrot48/jooq-ddl

**Comment by lukaseder** [comment]
Thanks for your report.

Ah, I see now. The other issue where you commented was about a `Meta` object that is based on generated code: https://github.com/jOOQ/jOOQ/issues/10594#issuecomment-1446200842

Your request is about the JDBC `DatabaseMetaData` backed `Meta` object. JDBC's `DatabaseMetaData` doesn't have API to expose source code for objects. We'd have to run additional queries in order to fetch those things. There's a feature request for this, already: https://github.com/jOOQ/jOOQ/issues/9818

It's probably not so hard to extract that information from the code generator logic. I'll look into implementing #9818 for jOOQ 3.18 right now.

**Comment by lukaseder** [comment]
Duplicate of #9818

**Comment by lukaseder** [comment]
Can confirm, https://github.com/jOOQ/jOOQ/issues/9818 will work quite simply for most dialects

**Comment by Rosenrot48** [comment]
Thanks! Sounds great! @lukaseder 

However, could you tell me will current fix be available in latest version (3.17) or only in dev version?

**Comment by lukaseder** [comment]
3.18 only.
